### PR TITLE
Fix isEmpty check for recursive bdd

### LINKED ISSUE #1041: Types containing only cyclic values are treated as empty
**Description:**

```ballerina
type R1 record {| R2 f; |};

type R2 record {| R1 f; |};

// @type R1 < L2
type L2 string[];

// @type R1 < I
type I int;

type RL1 [int, RL2];
// @type RL1 < I
type RL2 [int, RL1];

```

Subtyping relation between strictly recursive types seems to work as expected.

**Suggested Labels:**

**Suggested Assignees:**

**Affected Product Version:**

**OS, DB, other environment details and versions:**    

**Steps to reproduce:**

**Related Issues:**

### PR DESCRIPTION
**Pull request by lochana-chathura**
## Purpose
$subject.

Fixes #1041

### CODE REVIEW

File: compiler/modules/types/function.bal [hunk 1]
```diff
@@ -41,8 +41,7 @@ function functionSubtypeIsEmpty(Context cx, SubtypeData t) returns boolean {
         if res == () {
             // we've got a loop
             io:println("got a function loop");
-            // XXX is this right???
```
Comment (jclark) [inline compiler/modules/types/function.bal:44]: This is right: I have checked with Giuseppe.

File: compiler/testSuite/semtype/recursive-t.bal [hunk 2]
```diff
@@ -0,0 +1,14 @@
+type R1 record {| R2 f; |};
+
+type R2 record {| R1 f; |};
+
+// @type R1 <> L
+type L string[];
```
Comment (jclark) [inline compiler/testSuite/semtype/recursive-t.bal:6]: Do we currently fail this? They are different basic types...

Comment (lochana-chathura) [reply compiler/testSuite/semtype/recursive-t.bal:6]: Currently `// @type R1 < L` evaluates to `true`.  `R1^~L` becomes `R1` and when we check `isEmpty(R1)` that becomes `true` due to `if res == () { return true}` condition as in R1 we get the same bdd again when checking the emptiness.

Comment (jclark) [reply compiler/testSuite/semtype/recursive-t.bal:6]: So R1 and R2 are treated as equivalent to `never`?

Comment (lochana-chathura) [reply compiler/testSuite/semtype/recursive-t.bal:6]: At the moment that's the behaviour.

Comment (jclark) [reply compiler/testSuite/semtype/recursive-t.bal:6]: OK, so that's the real issue then.

Comment (jclark) [reply compiler/testSuite/semtype/recursive-t.bal:6]: Can you update issue accordingly?

Comment (jclark) [reply compiler/testSuite/semtype/recursive-t.bal:6]: Simple test case is then:

```
type L [L];
type N never;
// Should we have N == L
// @type N < L
```

File: compiler/modules/types/function.bal (hunk 1 above)
Comment (jclark) [reply compiler/modules/types/function.bal:44]: I am starting to wonder whether this fix might be the right one, given we have cells. I will ask.

**Comment by lochana-chathura** [comment]
Closing the PR since this is not the correct fix.

**Pull request closed without merging**